install/kubernetes/helm/istio/charts/sidecarInjectorWebhook/values.yaml

```diff
@@ -42,3 +42,18 @@ rewriteAppHTTPProbe: false
 neverInjectSelector: []
 
 alwaysInjectSelector: []
+
+# Enable objectSelector to filter out pods with no need for sidecar before calling istio-sidecar-injector.
+# It is disabled by default since this function will only work after k8s v1.15.
```

Comment:
seems very useful, wish we didn't have to disable it

Reply:
If enabled by default and the k8s version is less than v1.15, the `kubectl apply` will fail since kubectl will validate the spec and find that `objectSelector` is not supported.
For now, I think most of the k8s clusters may not meet the condition the have v1.15. We may enable this by default later?

Reply:
Yeah, I wish it was forward compatible and did nothing on older versions. Definitely should be default eventually, but not until we phase out support for 1.14 and below which will be a while. I think we aim to support 3 latest k8s versions

Reply:
`forward compatible` looks like a good reason. will enable this by default

Reply:
Sorry if I wasn't clear, I mean I wish Kubernetes would just not reject `objectSelector` and ignore it if it existed on old versions. I don't think we should enable by default since most users are not on 1.15 yet

Reply:
Sorry for my poor English... What I mean is that the kubectl will validate the spec locally.